### PR TITLE
Hopefully fixes the negative moles/heat bugs with fusion, as well as it occurring in fires.

### PR DESCRIPTION
**Pull request by as334**
Adds a check to fusion activation where if the purity is below 90% it won't activate, and adds various safety checks to the other parts of it to prevent negative mole/temperature issues.

### CODE REVIEW

**Comment by as334** [comment]
@duncathan

**Comment by KorPhaeron** [comment]
Let me know when this is merged and I'll update the server

File: code/datums/gas_mixture.dm [hunk 1]
```diff
@@ -133,15 +133,16 @@ What are the archived variables for?
...
+			carbon_dioxide = max(carbon_dioxide-carbon_catalyzed,0)
 			oxygen+=oxygen_added
 			nitrogen+=nitrogen_added
 			if(reaction_energy > 0)
 				reacting = 1
 				var/new_heat_capacity = heat_capacity()
 				if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
-					temperature = (temperature*old_heat_capacity + reaction_energy)/new_heat_capacity
+					var/newtemperature = (temperature*old_heat_capacity + reaction_energy)/new_heat_capacity
+					temperature=max(newtemperature,TCMB)//Prevents whatever mechanism is causing it to hit negative temperatures.
```
Comment (duncathan) [inline code/datums/gas_mixture.dm]: Replace these two lines with
`temperature = max(((temperature*old_heat_capacity + reaction_energy)/new_heat_capacity),TCMB)`

**Comment by as334** [comment]
@KorPhaeron

**Pull request merged**